### ISSUE TITLE
Theming: overriding is broken for gif images

### ISSUE THREAD

**Issue by VicDeo** [owncloud/core#31804]
1. Enable theme-example
2. Enable notifications app
3. Set `Mail Notifications` in personal page to 'Notify about all events'
3. php occ notifications:generate -u boo --link http://home.org  go home!

## Expected
Mail notification has `logo-email.gif` from theme-example

## Actual 
Mail notification has `logo-email.gif` from core

**Comment by michaelstingl** [comment]
@ChrisEdS FYI ^

**Comment by ownclouders** [comment]
[GitMate.io](https://gitmate.io) thinks possibly related issues are https://github.com/owncloud/core/issues/3209 (Theming), https://github.com/owncloud/core/issues/13402 (... broken), https://github.com/owncloud/core/issues/20484 (Orientation for shared images broken), https://github.com/owncloud/core/issues/31134 (Theming broken after trying to enabling a second theme), and https://github.com/owncloud/core/issues/23569 (Theming image path).

**Comment by PVince81** [comment]
affected version ? stable10 or 10.0.8 ? @VicDeo

**Comment by VicDeo** [comment]
@PVince81 I got this with master and didn't test further. But I guess @ChrisEdS had the same with OC 10

**Comment by ChrisEdS** [comment]
Yes @VicDeo - i've stumble upon this issue in 10.0.8

**Comment by VicDeo** [comment]
@PVince81 I'd like to start with this one  for the next sprint
also it would include `lib/private/URLGenerator.php` refactoring for https://github.com/owncloud/core/issues/31864

esimate: 1-1.5 md

**Comment by PVince81** [comment]
@VicDeo is the refactoring included in your estimate ? Or just the bits from #31864 required for the fix ?

**Comment by VicDeo** [comment]
@PVince81 yes, refactoring of lib/private/URLGenerator.php only is included as I suspect it to be the case
